varnishgather

```diff
@@ -428,6 +440,46 @@ done
 
 shift $((OPTIND - 1))
 
+if [ -z "$NAME" ]; then
+    VARNISHD_CMD=$(ps -o cmd= -p $(pgrep varnishd))
+    INSTANCE_NAME=$(echo "$VARNISHD_CMD" | grep -oP '(?<=-n )\S+')
+
+    if [ -n "$INSTANCE_NAME" ]; then
+        NAME="-n $INSTANCE_NAME"
+    else
+        INSTANCE_NAME=$(suggest_n_values)
+        if [ -n "$INSTANCE_NAME" ]; then
+            NAME="-n $INSTANCE_NAME"
+        fi
+    fi
+
+    if ! validate_instance; then
+        NAME=""
+    fi
+else
+    if ! validate_instance; then
+        INSTANCE_NAME=$(suggest_n_values)
+        if [ -n "$INSTANCE_NAME" ]; then
+            NAME="-n $INSTANCE_NAME"
+            if validate_instance; then
+                ID="$(cat /etc/hostname)-$(date +'%Y%m%d-%H%M%S')-$(n_opt "$INSTANCE_NAME")"
+                RELDIR="varnishgather-$ID"
+            else
+                NAME=""
+            fi
+        else
+            NAME=""
+        fi
+    fi
+fi
+
+if [ -n "$NAME" ] && [ -z "$RELDIR" ]; then
```

Comment:
this block feels overkill for me, I have the impression that you just need to do
```
RELDIR="$RELDIR-$NEWLYFIGUREDOUTNAME"
ID="$ID-$NEWLYFIGUREDOUTNAME"
```
if `NAME` wasn't specified
if

Reply:
I moved this into the block if block 439-452 after thinking about it I just reused INSTANCE_NAME set from get_varnishd_instance